libdnf/rpm/solv/map_iterator.hpp

```diff
@@ -38,25 +39,25 @@ class SolvMap;
 
 class SolvMapIterator {
 public:
-    explicit SolvMapIterator(const Map * map);
-    SolvMapIterator(const SolvMapIterator & other) = default;
+    explicit SolvMapIterator(const Map * map) noexcept;
+    SolvMapIterator(const SolvMapIterator & other) noexcept = default;
 
     using iterator_category = std::forward_iterator_tag;
-    using difference_type = PackageId;
+    using difference_type = std::ptrdiff_t;
     using value_type = PackageId;
     using pointer = void;
     using reference = void;
```

Comment:
Do you know why we use here a `void`?

Reply:
The `operator*()` usually returns a reference to an item in the internal container. It is usually declared as a `reference operator*();`. Our case is more complicated. Our `operator*()` returns a value. No a reference. The value is created as needed. I made a change. The `reference` type is defined in the same way as the `value_type`.  It looks confusing. However, it satisfies the assumption that `operator * ();` returns the `reference` type.
This is probably a better solution.

The `pointer` is used for `operator->()`.  In our case is more complicated. See above. So, we do not define `operator->()` and `void` is OK from my point of view.

Reply:
Thanks